examples/overlays/devenv.yaml

```
@@ -0,0 +1,9 @@
+inputs:
+  nixpkgs:
+    url: github:NixOS/nixpkgs/nixpkgs-unstable
+  rust-overlay:
+    url: github:oxalica/rust-overlay
+allowUnfree: true
+overlays:
+  rust-overlay:
+   - default
```

I don't want to uphold changes like this, but has `overlays` as part of the individual `inputs` also been considered? It seems less work and less error-prone to mention `rust-overlay` once.

```suggestion
inputs:
  nixpkgs:
    url: github:NixOS/nixpkgs/nixpkgs-unstable
  rust-overlay:
    url: github:oxalica/rust-overlay
    overlays:
      - default
allowUnfree: true
```

---

I thought of that, but what if Nix flakes suddenly also get this parameter? I'm afraid it could break devenv.

Maybe it's worth the risk and hope that it's done with the same API :)

---

Maybe

```
inputs:
  rust-overlay:
    devenv:
       overlays:
         - default
```

---

Or just not presume `inputs` is going to be passed through to Nix flakes. We have no idea what flakes are going to do before stabilization :sweat_smile:

If we explicitly support `inputs.<input>.url` and `inputs.<input>.follows`, then it is fine if thinks of more features, devenv first needs to be updated before those features can be used in devenv. It gives a bit more control over the `devenv.yaml` structure.

---

Implemented and also #185